### ISSUE TITLE
Restricted port number prohibits running mutliple instances of the bridge

### ISSUE THREAD

**Issue by JHannappel** [IBM/ibm-spectrum-scale-bridge-for-grafana#35]
**Description**
The port number argument is now limited to ony two choices, 4242 and 8443, where the 8443 port is used to select a TLS connection.
This makes it impossible to run multiple instances of the bridge on one node, because everyu instance needs it's own port.

**To Reproduce**
call zimonGrafanaIntf.py with -p 4243

**Expected behavior**
Being able to run the bridge with an arbitrary port, as it was possible in old versions.

**Desktop (please complete the following information):**
 We run several instances of the Bridge for our various clusters, most of them on one node.
 - IBM Spectrum Scale system type:
    - [*] IBM Spectrum Scale classic cluster.
    - [ ] IBM Spectrum Scale Container Native Storage Access (CNSA).
    - [*] IBM Elastic Storage Server (ESS).
 - IBM Spectrum Scale cluster version: 5.0.5.1
 - IBM Spectrum Scale bridge for Grafana version: 6.1.3
 - Grafana version: 8.0.3
 - OS Grafana is installed on: CentOS 7

**Additional context**

**Comment by Helene** [comment]
what do you mean with the node? gpfs node? host?

**Comment by JHannappel** [comment]
Hello, 
I mean the node that's used to run the grafana bridge. 
In our setup we have 15 GPFS clusters and we have only a few 
machines suitable for running the grafana bridge, due to various reasons like load, 
netweork conectivity and so on, so we have to run multiple instances of he bridge on 
one single node. This is now a gpfs node which runs alo one of the PMcollectors. 

Greetings, 
Jürgen

**Comment by Helene** [comment]
Hello Juergen,
thanks for your quick response. One question more: How the bridges are connected to the GPFS pmcollector nodes, I guess remotely? And 1 bridge per 1 pmcollector per GPFS cluster, or do you have multiple bridges connected to the same gpfs cluster?

**Comment by JHannappel** [comment]
Hello, 
various. One bridge is directly (localhost) connected to the pmcollector, the others are remote connected. 
We also have for the biggest cluster 2 bridges, because we saw in the past performance problems when 
many grafana browser tabs were open ("many" meaning ~4), so we have two grafana instances hat can access the 
data (one for the peaople at european XFEL, one for the crew at DESY) and we spent one bridge instance for each grafana instance.

**Comment by Helene** [comment]
thanks Juergen, for giving me some updates about your environment. I'm thinking about removing fix port bindings for http, but https connections as well, from the source code and add https selection controll over additional flaq (option). It might take a couple of days before I'll be ready with the fix . As soon as the fix is done, I'll publish the new 6.1.4 version. As workaround for you, edit the [line](https://github.com/IBM/ibm-spectrum-scale-bridge-for-grafana/blob/master/source/confParser.py#L181) in the source code as follows:
```
parser.add_argument('-p', '--port', action="store", type=int, default=None, help='port number listening on for HTTP(S) connections (Default from config.ini: 4242)')
```
...and restart the bridge using -p 4243

**Comment by JHannappel** [comment]
Thanks! 
I applied such a patch already and the bridges are running now. 

A side remark: 
Am I right that the support for 5.1.1 with the apiKeys is only available in the Bridge v7 branch?

**Comment by Helene** [comment]
Right, the apiKeys have been introduced with GPFS release code 5.1.1. For those, the bridge 7.0 need to be used. The port changes will be merged in the 7.x as well.

**Comment by Helene** [comment]
This request have been resolved with the commit [51425587f68ad9c974c22b55e992fcd21465d462](https://github.com/IBM/ibm-spectrum-scale-bridge-for-grafana/commit/51425587f68ad9c974c22b55e992fcd21465d462)

The fix is available  in the bridge release version [6.1.4](https://github.com/IBM/ibm-spectrum-scale-bridge-for-grafana/releases/tag/v6.1.4) and will be included in the version 7.0.1 as well.
